src/games.c

```diff
@@ -1691,7 +1691,28 @@ int igraph_callaway_traits_game (igraph_t *graph, igraph_integer_t nodes,
     igraph_real_t maxcum;
     igraph_vector_t nodetypes;
 
-    /* TODO: parameter checks */
+    //parameter check 
+    if(nodes < 0){
+        IGRAPH_ERROR("Invalid number of vertices", IGRAPH_EINVAL);
```

Comment:
It's best to be clear about _why_ a value is invalid. I would suggest "The number of vertices must be non-negative". Of course, it is obvious that it should be non-negative, but this clearly communicates to the user that all non-negative values are acceptable.